chapters/06-smart-contracts-in-ralph.md

```diff
@@ -1554,53 +1556,59 @@ import { AssetFunctions } from '../artifacts/ts'
 import { Address, DUST_AMOUNT, NodeProvider } from '@alephium/web3'
 
 async function test() {
+  const nodeProvider = new NodeProvider('http://127.0.0.1:22973')
   const signer = await getSigner()
-  const nodeProvider = new NodeProvider("http://localhost:22973")
 
   const { contractInstance: asset } = await AssetFunctions.deploy(signer, {
-     initialFields: { owner: signer.address, n: 100n },
-     issueTokenAmount: 100n,
-     issueTokenTo: signer.address
+    initialFields: { owner: signer.address, n: 100n },
+    issueTokenAmount: 100n,
+    issueTokenTo: signer.address
   })
 
-  async function getBalance(address: Address) {
+  async function getTokenBalance(address: Address) {
     const balance = await nodeProvider.addresses.getAddressesAddressBalance(address)
     console.assert(balance.tokenBalances?.[0].id === asset.contractId)
-    return balance
+
+    return {
+      totalBalance: balance.tokenBalances?.[0].amount,
+      lockedBalance: balance.lockedTokenBalances?.[0].amount
+    }
   }
 
   await asset.transact.burn({
-     signer,
-     tokens: [{ id: asset.contractId, amount: 1n }]
+    signer,
+    tokens: [{ id: asset.contractId, amount: 1n }]
   })
-  const balance = await getBalance(signer.address)
-  console.assert(balance.tokenBalances?.[0].amount === "99")
+  const signerBalance = await getTokenBalance(signer.address)
+  console.assert(signerBalance.totalBalance === '99')
 
   await asset.transact.lock({
-     signer,
-     args: { amount: 1n },
-     attoAlphAmount: DUST_AMOUNT,
-     tokens: [{ id: asset.contractId, amount: 1n }]
+    signer,
+    args: { amount: 1n },
+    attoAlphAmount: DUST_AMOUNT,
+    tokens: [{ id: asset.contractId, amount: 1n }]
   })
-  const balance2 = await getBalance(signer.address)
-  console.assert(balance2.tokenBalances?.[0].amount === "99")
-  console.assert(balance2.lockedTokenBalances?.[0].amount === "1")
+  const signerBalance2 = await getTokenBalance(signer.address)
+  console.assert(signerBalance2.totalBalance === '99')
+  console.assert(signerBalance2.lockedBalance === '1')
 
   await asset.transact.deposit({
     signer,
     args: { amount: 1n },
     tokens: [{ id: asset.contractId, amount: 1n }]
   })
-  const balance3 = await getBalance(signer.address)
-  console.assert(balance3.tokenBalances?.[0].amount === "98")
+  const signerBalance3 = await getTokenBalance(signer.address)
+  console.assert(signerBalance3.totalBalance === '98')
+  const contractBalance = await getTokenBalance(asset.address)
+  console.assert(contractBalance.totalBalance === '1')
```

Comment:
I added an assertion for the contract state to demonstrate that the `deposit` call took 1 token from the signer and moved it to the contract state. Let me know if you find this useful!